static/app/components/commandPalette/ui/list.tsx

```diff
@@ -0,0 +1,213 @@
+  const {collectionProps} = useSelectableCollection({
+    selectionManager: treeState.selectionManager,
+    keyboardDelegate: new ListKeyboardDelegate({
+      collection: treeState.collection,
+      disabledKeys: treeState.selectionManager.disabledKeys,
+      ref: inputRef,
+    }),
+    shouldFocusWrap: true,
+    ref: inputRef,
+  });
+
+  const inputProps = mergeProps(collectionProps, {
+    onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+      setQuery(e.target.value);
+      // We want to reset the focused key when the user types.
+      // The useLayoutEffect above will ensure that we set it correctly when it becomes null
+      treeState.selectionManager.setFocusedKey(null);
+    },
+    onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === 'Backspace' && query === '') {
+        clearSelection();
+        e.preventDefault();
+      }
+
+      if (e.key === 'Enter' || e.key === 'Tab') {
+        onActionKey(treeState.selectionManager.focusedKey);
+      }
+    },
+  });
+
+  const placeholder = useMemo(() => {
+    if (selectedAction) {
+      return selectedAction.label;
+    }
+    return t('Type for actions…');
+  }, [selectedAction]);
+
+  return (
+    <Fragment>
+      <InputHeader>
+        <SearchInputContainer>
+          {selectedAction && (
+            <Button
+              borderless
+              size="sm"
+              icon={<IconArrow direction="left" />}
+              onClick={() => {
+                clearSelection();
+                inputRef.current?.focus();
+              }}
+              aria-label={t('Return to all options')}
+            />
+          )}
+          <CommandInput
+            ref={inputRef}
+            value={query}
+            aria-label={t('Search commands')}
+            placeholder={placeholder}
+            autoFocus
+            {...inputProps}
+          />
+        </SearchInputContainer>
+      </InputHeader>
+      {treeState.collection.size === 0 ? (
+        <EmptyWrap>
+          <img src={error} alt="No results" />
+          <p>Whoops… we couldn't find any results matching your search.</p>
+          <p>Try rephrasing your query maybe?</p>
+        </EmptyWrap>
+      ) : null}
+      <ResultsList>
+        <ListBox
+          listState={treeState}
+          keyDownHandler={() => true}
+          overlayIsOpen
+          size="md"
+          aria-label="Search results"
+          selectionMode="none"
+          onAction={key => onActionKey?.(key)}
+          shouldUseVirtualFocus
+        />
+      </ResultsList>
+    </Fragment>
+  );
+}
+
+const InputHeader = styled('div')`
+  display: flex;
+  flex-direction: column;
+  align-items: start;
+  box-shadow: 0 1px 0 0 ${p => p.theme.translucentInnerBorder};
+  z-index: 2;
+`;
+
+const SearchInputContainer = styled('div')`
+  position: relative;
+  display: flex;
+  align-items: center;
+  padding: 0 ${p => p.theme.space.md};
+  width: 100%;
+`;
+
+const CommandInput = styled('input')`
```

Comment:
Can we reuse the input component, and if not, what would we need to change to make that happen? :D?